src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+        }
+
+        private interface ISarifLogBaseliner
+        {
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif, bool assignGuidsToNewResults);
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif);
+        }
+
+        private class SarifLog
+        {
+            public IList<Run> Runs { get; internal set; }
+
+            internal static IEnumerable<WorkItemFilingMetadata> CreateWorkItemFilingMetadataFromLogs(IEnumerable<SarifLog> sarifLogsToFile)
+            {
+                throw new NotImplementedException();
+            }
+
+            public class Run
+            {
+                public IList<Result> Results { get; internal set; }
+            }
+        }
+
+        private class SarifLogBaselinerFactory
+        {
+            internal static ISarifLogBaseliner CreateSarifLogBaseliner(object standard)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class SarifBaselineType
+        {
+            public static object Standard { get; internal set; }
+        }
+
+        private class SarifSplitter
+        {
+            internal static IEnumerable<SarifLog> Split(SarifLog sarifLog, Func<Result, bool> filterPredicate, SplittingStrategy splittingStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            internal class Result
+            {
+                public BaselineState BaselineState;
+            }
+
+            internal class SarifSplittingStrategy
+            {
+            }
+        }
+
+        private enum BaselineState
+        {
+            New
+        }
+
+        private enum SplittingStrategy
+        {
+            OneFilePerRunPerRule
+        }
+
+        private class WorkItemFilingMetadata
+        {
+            public string Title { get; internal set; }
+
+            public string AreaPath { get; internal set; }
+
+            public string Body { get; internal set; }
+
+            public HashSet<string> Tags { get; internal set; }
+
+            public Uri Uri { get; internal set; }
+
+            public List<object> Attachments { get; internal set; }
+        }
+
+
+        private static void LogFailure(Exception exception)
+        {
+            throw new NotImplementedException();
+        }
+
+        private class AdoWorkItemFiler
+        {
+            public AdoWorkItemFiler()
+            {
+            }
+
+            public bool FileAttachments { get; internal set; }
+
+            public event EventHandler<WorkItemFiledOrUpdatedArgs> WorkItemFiledOrUpdated;
+
+            internal async Task FileWorkItems(IEnumerable<WorkItemFilingMetadata> workItemsToFile)
+            {
+                throw new NotImplementedException();
+            }
+
+            public class WorkItemFiledOrUpdatedArgs
+            {
+                public WorkItemFilingMetadata Metadata;
+            }
```

Comment:
>} [](start = 12, length = 1)

YAGNI, but at some point we might find we want a flag to tell whether it was filed or updated.

Reply:
Agreed. We haven't really dug into the contents of the metadata class because this API usability study is strictly around 'file all new bugs'. A next use case might add 'Updated' to compare/contrast. A further use case would consider 'close resolved items'. I actually think that latter case is most interesting to us in the short-term.

---
In reply to: [307087950](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307087950) [](ancestors = 307087950)